### PR TITLE
ENH: Add OpenMEEG for 3-layer MEG and EEG

### PR DESCRIPTION
**Pull request by larsoner**
~~No real support here yet, just checking first to make sure that adding these to `requirements.txt` and `environment.yml` doesn't break CIs. Then eventually we can add proper support following https://github.com/mne-tools/mne-openmeeg or something similar~~

- [x] Add OpenMEEG to CIs on PyPi and conda-forge
- [x] Get green
- [x] Add BEM shims for `method='openmeeg'`
- [x] Actually incorporate OpenMEEG code in the two TODO places (@agramfort)
- [x] Release OpenMEEG 2.5.5 with all fixes
- [x] Require it in `environment.yml` and `requirements.txt`
- [x] Do something about `comps`
- [x] Do something about windows
- [ ] ~~Do something about the required test time~~

Helps with #1624

### CODE REVIEW

**Comment by larsoner** [comment]
Okay looks like the failures thus far are just due to an `eeglabio` update. I'll restart this once we figure out how to deal with that

**Comment by larsoner** [comment]
macOS failure is probably https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/issues/310, a bug a caused by me messing up our recipe... hopefully I can fix it soon

**Comment by larsoner** [comment]
@agramfort I added your shims for OpenMEEG-based BEM, I/O, and forward computations, with some tests that you can use when you start trying to port over code. PyPi and conda-forge are updated:

https://pypi.org/project/openmeeg/2.5.1/
https://anaconda.org/conda-forge/openmeeg

Now you can just have fun :)

**Comment by larsoner** [comment]
@agramfort Windows is... having trouble with this. Can you write a little test over in the OpenMEEG repo using equivalent files? Or would it help for me to give it a shot? It'll be easier to debug over there.

**Comment by larsoner** [comment]
@agramfort for now I'm inclined to say it only works on windows, and raise an error there if you try to do it. That way we can continue to make progress on the sEEG generalization aspects in follow-up PRs. WDYT?

**Comment by larsoner** [comment]
... and this is painful:
```
56.70s call     mne/forward/tests/test_make_forward.py::test_make_forward_solution_openmeeg[3]
```
but I'm not sure there is much we can do about it :(

**Comment by larsoner** [comment]
... and we have a problem that `comps` is never used here

**Comment by larsoner** [comment]
Looking at the code, I think the best way to handle compensation is with a heavy refactoring of the forward computation code. I could do this at the same time as the MEG+EEG -> MEG+EEG+... dict-based approach that will facilitate iEEG channel forward computation (and maybe even `fNIRS` someday!).

My vote is to actually get that done and merged *first*, then rebase this PR and get it to fit within that framework.

**Comment by agramfort** [comment]
+1

**Comment by larsoner** [comment]
Got it to fail on Windows 3.10 pip pre. Let's try with 2.6.0.dev1 from TestPyPi which has OpenMP disabled...

**Comment by larsoner** [comment]
@agramfort do you want to attempt the rebase here or would you like me to do it?

**Comment by agramfort** [comment]
I won't have time in the coming days in the next days.

File: mne/forward/tests/test_make_forward.py [hunk 1]
```diff
@@ -186,41 +225,76 @@ def test_make_forward_solution_bti(fname_src_small):
...
+        fname_ctf_raw, fname_trans, src, fname_bem_meg, eeg=False,
+        mindist=mindist, verbose=True)
+
+    if other == 'openmeeg':
+        # TODO: This should be a 1-layer, but it's broken
+        # (some correlations become negative!)...
+        bem_surfaces = read_bem_surfaces(fname_bem)  # fname_bem_meg
+        bem = make_bem_solution(bem_surfaces, solver='openmeeg')
+        # TODO: These tolerances are bad
+        tol_kwargs = dict(meg_atol=1, meg_corr_tol=0.65, meg_rdm_tol=0.6)
```
Comment (larsoner) [inline mne/forward/tests/test_make_forward.py:255]: @agramfort I added a test for CTF data. The results there are less good, but maybe okay? (I had to use a 3-layer as the 1-layer is also terrible here as well, some correlations in that case are negative...)

Comment (agramfort) [reply mne/forward/tests/test_make_forward.py:255]: it's possible the bug in Python wrapper with ref / shared pointers affects our results here. I fear we need to fix it there first. Some alternative to move faster here is to work by writing files on disk and reading them without numpy arrays

Comment (larsoner) [reply mne/forward/tests/test_make_forward.py:256]: Using files for the geom at least does not help with these tols

**Comment by larsoner** [comment]
Okay with file-based IO for `_make_openmeeg_geometry` we get on Windows:
```
E   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\openmeeg-io-t2wktag7\\tmp.geom'
```
this suggests that OpenMEEG is not letting go of the file pointer. I added this to the OpenMEEG issue tracker

**Comment by larsoner** [comment]
@agramfort feel free to merge if you're happy

**Review by agramfort — APPROVED** [review]
🙏 🙏 🙏  @larsoner 

this is a big one 🚀

**Pull request merged**